scheduler/roles/slurm/tasks/_config_files.yml

```diff
@@ -0,0 +1,117 @@
+# Copyright 2025 Dell Inc. or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Append share path if NFS
+  ansible.builtin.set_fact:
+    slurm_share_prefix: "{{ share_path }}/{{ slurm_dir_name }}"
+
+- name: Ensure Slurm conf share dir
+  ansible.builtin.file:
+    path: "{{ slurm_share_prefix }}{{ slurm_config_dir }}"
+    state: directory
+    owner: root
+    group: root
+    mode: "{{ common_mode }}"
+
+- name: Ensure directories for slurm ops
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ slurm_user }}"
+    group: "{{ slurm_user_group }}"
+    mode: "{{ common_mode }}"
+  loop:
+    - "{{ slurm_conf_dict['StateSaveLocation'] }}"
+    - "{{ slurm_config_dir }}"
+
+- name: Create slurm.conf
+  ansible.builtin.template: # Create dest directory before
+    src: "slurm.conf.j2"
+    dest: "{{ slurm_share_prefix }}{{ slurm_config_dir }}/slurm.conf"
+    # owner: root
+    # group: root
+    mode: "{{ conf_file_mode }}"
+  run_once: true
+  notify:
+    - Restart slurmctld
+
+- name: Create cgroup.conf
+  ansible.builtin.template: # Create dest directory before
+    src: "cgroup.conf.j2"
+    dest: "{{ item }}"
+    owner: root
+    group: root
+    mode: "{{ conf_file_mode }}"
+  loop:
+    - "{{ slurm_share_prefix }}{{ slurm_config_dir }}/cgroup.conf"
+    - "{{ slurm_config_dir }}/cgroup.conf"
+  notify:
+    - Restart slurmctld
+    - Restart slurmdbd
+
+# Environment Var SLURM_CONF
+- name: Set env variable
+  community.general.ini_file:
+    path: "/etc/environment"
+    option: SLURM_CONF
+    value: "{{ slurm_share_prefix }}{{ slurm_config_dir }}/slurm.conf"
+    no_extra_spaces: true
+    mode: u=rw,g=r,o=r
```

Comment:
mode should be used in format of number. 644

Reply:
we wanted to follow a common template across omnia and use mode as variable. @priti-parate @snarthan can you confirm